cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/LedgerEvents/Gov/PredefinedAbstainDRep.hs

```diff
@@ -0,0 +1,362 @@
+  H.note_ $ "Socketpath: " <> socketPath
+  H.note_ $ "Foldblocks config file: " <> configurationFile
+
+  gov <- H.createDirectoryIfMissing $ work </> "governance"
+
+  initialDesiredNumberOfPools <- getDesiredPoolNumberValue epochStateView ceo
+
+  let newNumberOfDesiredPools = initialDesiredNumberOfPools + 1
+
+  -- Do some proposal and vote yes with the first DRep only
+  -- and assert that proposal does NOT pass.
+  void $ desiredPoolNumberProposalTest execConfig epochStateView configurationFile socketPath ceo gov "firstProposal"
+                                       wallet0 Nothing [(1, "yes")] newNumberOfDesiredPools 3 (Just initialDesiredNumberOfPools) 10
+
+  -- Take the last two stake delegators and delegate them to "Abstain".
+  delegateToAlwaysAbstain execConfig epochStateView configurationFile socketPath sbe gov "delegateToAbstain1"
+                          wallet1 (defaultDelegatorStakeKeyPair 2)
+  delegateToAlwaysAbstain execConfig epochStateView configurationFile socketPath sbe gov "delegateToAbstain2"
+                          wallet2 (defaultDelegatorStakeKeyPair 3)
+
+  -- Do some other proposal and vote yes with first DRep only
+  -- and assert the new proposal passes now.
+  let newNumberOfDesiredPools2 = newNumberOfDesiredPools + 1
+  void $ desiredPoolNumberProposalTest execConfig epochStateView configurationFile socketPath ceo gov "secondProposal"
+                                       wallet0 Nothing [(1, "yes")] newNumberOfDesiredPools2 0 (Just newNumberOfDesiredPools2) 10
+
+delegateToAlwaysAbstain
+  :: (HasCallStack, MonadTest m, MonadIO m, H.MonadAssertion m, MonadCatch m)
+  => H.ExecConfig -- ^ Specifies the CLI execution configuration.
+  -> EpochStateView -- ^ Current epoch state view for transaction building. It can be obtained
+                    -- using the 'getEpochStateView' function.
+  -> FilePath -- ^ Path to the node configuration file as returned by 'cardanoTestnetDefault'.
+  -> FilePath -- ^ Path to the cardano-node unix socket file.
+  -> ShelleyBasedEra ConwayEra -- ^ The Shelley-based era (e.g., 'ConwayEra') in which the transaction will be constructed.
+  -> FilePath -- ^ Base directory path where generated files will be stored.
+  -> String -- ^ Name for the subfolder that will be created under 'work' folder.
+  -> PaymentKeyInfo -- ^ Wallet that will pay for the transaction.
+  -> StakingKeyPair -- ^ Staking key pair used for delegation.
+  -> m ()
+delegateToAlwaysAbstain execConfig epochStateView configurationFile socketPath sbe work prefix
+                        payingWallet skeyPair@(StakingKeyPair vKeyFile _sKeyFile) = do
+
+  let era = toCardanoEra sbe
+      cEra = AnyCardanoEra era
+
+  baseDir <- H.createDirectoryIfMissing $ work </> prefix
+
+  -- Create vote delegation certificate
+  let voteDelegationCertificatePath = baseDir </> "delegation-certificate.delegcert"
+  void $ H.execCli' execConfig
+    [ "conway", "stake-address", "vote-delegation-certificate"
```

Comment:
Don't hardcode the era otherwise we create more work for us in the future when we go beyond conway. Use `anyEraToString`.

Reply:
Fixed here: https://github.com/IntersectMBO/cardano-node/pull/5815/commits/7c5cd80cf6b496b7fe2445825810662858423180